rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1850,6 +1913,179 @@ public Response deleteStudentContactLogEntry(@PathParam("STUDENTID") Long studen
 
     return Response.noContent().build();
   }
+  
+  @Path("/students/{STUDENTID:[0-9]*}/contactLogEntry/{ID:[0-9]*}/entryComments")
+  @GET
+  @RESTPermit(StudentContactLogEntryPermissions.FIND_STUDENTCONTACTLOGENTRY)
+  public Response listStudentContactLogEntryCommentsById(@PathParam("STUDENTID") Long studentId, @PathParam("ID") Long id) {
+    Student student = studentController.findStudentById(studentId);
+    Status studentStatus = checkStudent(student);
+    if (studentStatus != Status.OK)
+      return Response.status(studentStatus).build();
+
+    StudentContactLogEntry contactLogEntry = studentContactLogEntryController.findContactLogEntryById(id);
+    if (contactLogEntry == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    if (contactLogEntry.getArchived()) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    if (!contactLogEntry.getStudent().getId().equals(contactLogEntry.getStudent().getId())) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    User loggedUser = sessionController.getUser();
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    Boolean amIGuidanceCounselor = studentController.amIGuidanceCouncelor(studentId, staffMember);
+
+    if (!loggedUser.getRole().equals(Role.ADMINISTRATOR)) {
+      if (!amIGuidanceCounselor) {
+      // If logged user is teacher, return only their own entries and comments under
+        if (loggedUser.getRole().equals(Role.TEACHER)) {
+          List<StudentContactLogEntryComment> ownEntries = new ArrayList<>();
+          List<StudentContactLogEntryComment> entryComments = studentContactLogEntryCommentController.listContactLogEntryCommentsByEntry(contactLogEntry);
+          
+          for (StudentContactLogEntryComment entry : entryComments) {
+            if (entry.getEntry().getCreatorId() != null) {
+              if (entry.getEntry().getCreatorId().equals(loggedUser.getId())){
+                ownEntries.add(entry);
+              } 
+            }
+          }
+          
+          return Response.ok(objectFactory.createModel(ownEntries)).build();
+        } else {
+          return Response.status(Status.FORBIDDEN).build();
+        }
+      }
+    }
+    return Response.ok(objectFactory.createModel(studentContactLogEntryCommentController.listContactLogEntryCommentsByEntry(contactLogEntry))).build();
+
+  }
+  
+  @Path("/students/{ID:[0-9]*}/contactLogEntry/entryComments")
```

Comment:
Shouldn't this be more like...

`@Path("/students/{STUDENTID:[0-9]*}/contactLogEntries/{ENTRYID:[0-9]*}/comments")`